### PR TITLE
feat(labware-creator): support multi-channel pipettes

### LINKED ISSUE #7965: Multi-channel compatibility features in LC
## Overview

Allow using multi-channel pipettes in the test protocol, when the labware def is compatible with multi-channel.

## Design
https://www.figma.com/file/9vGYxqlhdukz2wFDbGNqE5/Labware-Library-and-Creator?node-id=529%3A25

## Acceptance Criteria
- Multi-channel options appear in Pipette dropdown

- When generated labware definition isn't compatible with labware, disable the multi-channel options
- Disabled multi-channel options get a popover: "Labware is incompatible with 8-Channel pipettes)

- When labware def can't be generated (eg missing parameters), the whole pipette field is disabled
- When the pipette field is disabled, the whole field has the popover "Add missing measurements to select a test pipette")

- Making changes to any measurement fields will clear out the pipette field (use a list of the fields that don't affect measurement, as this will be less fragile than a list of fields that DO affect measurement. Eg brand is sure to NOT affect measurement, so that's safe to avoid updating Pipette field when changed)

### PR DESCRIPTION
**Pull request by IanLondon**
# Overview

Closes #7965

# Changelog

- Instead of generating a "defaulted definition" inside the `ConditionalRender` component, this PR moves that process into the main Formik child, and stores the defaulted definition in Formik state (called `status`) so that it can be used in other components. (FYI, `status` is a special place in Formik for storing arbitrary state stuff)
- Goes and uses that definition in the `status` state to control the Pipette field

# Review requests

- [ ] Missing any test coverage? Eg Export settings Pipettes field might have some uncovered cases...
- [x] Multi-channel options appear in Pipette dropdown

- [x] When generated labware definition isn't compatible with labware, disable the multi-channel options
- [x] Disabled multi-channel options get a popover: "Labware is incompatible with 8-Channel pipettes)

- [x] When labware def can't be generated (eg missing parameters), the whole pipette field is disabled
- [x] When the pipette field is disabled, the whole field has the popover "Add missing measurements to select a test pipette")

- [x] Making changes to any measurement fields will clear out the pipette field (use a list of the fields that don't affect measurement, as this will be less fragile than a list of fields that DO affect measurement. Eg brand is sure to NOT affect measurement, so that's safe to avoid updating Pipette field when changed)

# Risk assessment

Med-high, but LC only

### CODE REVIEW

File: labware-library/src/labware-creator/components/Dropdown.tsx [hunk 1]
```diff
@@ -34,36 +37,52 @@ export const Dropdown = (props: DropdownProps): JSX.Element => {
...
+        <label
+          className={cx(fieldStyles.field_label, {
+            [fieldStyles.disabled]: props.disabled,
+          })}
+        >
+          <Field name={props.name}>
+            {/* @ts-expect-error(IL, 2021-03-24): formik types need cleanup w LabwareFields */}
+            {({ field, form }) => (
+              <div style={{ width: '18rem' }}>
+                {/* TODO IMMEDIATELY ^^^ don't inline style; allow instance to be styled via style props */}
```
Comment (IanLondon) [inline labware-library/src/labware-creator/components/Dropdown.tsx]: oops missed this

File: labware-library/src/labware-creator/components/sections/Export.tsx [hunk 2]
```diff
@@ -23,7 +24,10 @@ interface ExportProps {
 
 export const Export = (props: ExportProps): JSX.Element | null => {
   const fieldList: Array<keyof LabwareFields> = ['pipetteName']
-  const { values, errors, touched } = useFormikContext<LabwareFields>()
+  const _context = useFormikContext<LabwareFields>()
+  const { values, errors, touched } = _context
+  const status: FormStatus = _context.status
```
Comment (IanLondon) [inline labware-library/src/labware-creator/components/sections/Export.tsx:29]: annoyingly, `Formik` and `useFormikContext` don't allow `status` to be typed via generics, so it has to be explicitly typed like this each time, which makes for some annoying syntax changes in this PR. But it's better than leaving it `any`-typed!

File: labware-library/src/labware-creator/fields.ts [hunk 3]
```diff
@@ -350,6 +363,11 @@ export const getImplicitAutofillValues = (
   return result
 }
 
+export const getInitialStatus = (): FormStatus => ({
+  defaultedDef: null,
```
Comment (IanLondon) [inline labware-library/src/labware-creator/fields.ts:367]: "defaultedDef" is the best we could come up with for representing "a labware definition with all the fields that don't affect XY geometry filled in with arbitrary defaults, for the purpose of displaying the conditional labware render and for determining multi-channel compatibility." 🙃 🤯

**Review by jerader — COMMENTED** [review]
Testing a few different cases and found these errors:
- if you select 96 well aluminum block + tube, the multichannel pipettes are still available, this should not be the case since I interpret that as a singlePCR tube on the block (this is different from the PCR strip tube option)
- this sequence shouldn't be allowed i don't think:
1. fill in the labware info, making in compatible with a multichannel
2. select multichannel
3. go back and change labware criteria to be incompatible with multichannel
4. the multichannel is still selected and is downloadable, even though the criteria should no longer fit. 
Is that something we want or do we encourage customers to refresh the page?


All the other review requests look good to me.

**Review by jerader — APPROVED** [review]
looks good! tested a few different labwares:

- 96 well PCR plate
- 24 tube rack
- reservoir with less than 8 rows
- 96 well aluminum block
- tip rack
- well plate with less than 8 rows
- 1 well reservoir with NEST dimensions
- 1 well reservoir with small dimensions - 60 x 40 mm (to make multichannel incompatible)

**Pull request merged**